currency_rate_update/models/res_currency_rate_provider.py

```diff
@@ -156,8 +157,10 @@ def _update(self, date_from, date_to, newest_only=False):
                 continue
 
             if not data:
-                # Try again if there is no data yet
+                if provider.no_data_skip:
+                    provider._schedule_next_run()
```

Comment:
```suggestion
                if provider.force_update_next_run:
                    provider.next_run = fields.Date.context_today(self)
```

We should not go as far as running _schedule_next_run() if there is no data.  We should just update next_run to today.